dallinger/custom.py

```diff
@@ -1234,6 +1236,7 @@ def worker_function(event_type, assignment_id, participant_id):
                 if not attended:
                     exp.log("Attention check failed.", key)
                     participant.status = "did_not_attend"
+                    session.commit()
                     exp.attention_check_failed(participant=participant)
                     exp.recruiter().recruit_participants(n=1)
                 else:
```

Comment:
finally, we really need a commit prior to exp.recruit on line 1244/1247 - otherwise we are requiring the experimenter to add commits within their recruit method